Client/Telemetry/UnifiedTelemetry.swift

```diff
@@ -0,0 +1,95 @@
+//    public static let click = "click"
+//    public static let change = "change"
+//    public static let open = "open"
+//    public static let openAppStore = "open_app_store"
+//    public static let share = "share"
+//    public static let swipeToNavigateBack = "swipe_to_navigate_back"
+//    public static let swipeToNavigateForward = "swipe_to_navigate_forward"
+}
+
+class TelemetryEventObject {
+    public static let app = "app"
+//    public static let searchBar = "search_bar"
+//    public static let eraseButton = "erase_button"
+//    public static let settingsButton = "settings_button"
+//    public static let setting = "setting"
+//    public static let menu = "menu"
+//    public static let pasteAndGo = "paste_and_go"
+}
+
+class UnifiedTelemetry {
+
+    init(profile: Profile) {
+        NotificationCenter.default.addObserver(self, selector: #selector(UnifiedTelemetry.appWillResignActive(notification:)), name: NSNotification.Name.UIApplicationWillResignActive, object: nil)
+
+        NotificationCenter.default.addObserver(self, selector: #selector(UnifiedTelemetry.appDidEnterBackground(notification:)), name: NSNotification.Name.UIApplicationDidEnterBackground, object: nil)
+
+        NotificationCenter.default.addObserver(self, selector: #selector(UnifiedTelemetry.appDidBecomeActive(notification:)), name: NSNotification.Name.UIApplicationDidBecomeActive, object: nil)
+
+        setupCorePing(withProfile: profile)
+    }
+
+    @objc func appWillResignActive(notification: NSNotification) {
+        Telemetry.default.recordSessionEnd()
+    }
+
+    @objc func appDidEnterBackground(notification: NSNotification) {
+        Telemetry.default.queue(pingType: CorePingBuilder.PingType)
+        Telemetry.default.scheduleUpload(pingType: CorePingBuilder.PingType)
+    }
+
+    @objc func appDidBecomeActive(notification: NSNotification) {
+        Telemetry.default.recordSessionStart()
+        Telemetry.default.recordEvent(category: TelemetryEventCategory.action, method: TelemetryEventMethod.foreground, object: TelemetryEventObject.app)
+    }
+
+    fileprivate func setupCorePing(withProfile profile: Profile) {
+        let telemetryConfig = Telemetry.default.configuration
+        telemetryConfig.appName = AppInfo.displayName
+        telemetryConfig.userDefaultsSuiteName = AppInfo.sharedContainerIdentifier
+        telemetryConfig.dataDirectory = .documentDirectory
+
+        #if DEBUG
+            telemetryConfig.updateChannel = "debug"
+            telemetryConfig.isCollectionEnabled = false
+            telemetryConfig.isUploadEnabled = false
+        #else
+            telemetryConfig.updateChannel = "release"
+            let sendUsageData = profile.prefs.boolForKey(AppConstants.PrefSendUsageData) ?? true
+            telemetryConfig.isCollectionEnabled = sendUsageData
+            telemetryConfig.isUploadEnabled = sendUsageData
+        #endif
+
+
+        // TODO: Add these items at ping root
```

Comment:
@justindarc I'll ask you about hooking these up

Reply:
Wouldn't these just be:

```
telemetryConfig.measureUserDefaultsSetting(forKey: NewTabAccessors.PrefKey, withDefaultValue: profile.prefs.stringForKey(NewTabAccessors.PrefKey))
telemetryConfig.measureUserDefaultsSetting(forKey: PrefsKeys.KeyMailToOption, withDefaultValue: profile.prefs.stringForKey(PrefsKeys.KeyMailToOption))
```

???

Reply:
Note that `UserDefaults` will get re-captured at the time we queue the ping to be sent. The only thing we're doing when calling `measureUserDefaultsSetting()` is letting telemetry-ios know that "these are the `UserDefaults` keys we're interested in" and supplying them with default values (in case the `UserDefaults` haven't been inited yet)

Reply:
Ok I thought it had to be like this to appear in the root correctly `telemetryConfig.defaultNewTabExperience`

Reply:
D'oh.. wait.. looks like `UserDefaults` aren't measured in the CorePing currently... that can change though.

In telemetry-ios, any measured `UserDefaults` values appear under `"settings": { ... }` in the `FocusEventPing`

Reply:
I am not sure we have approval to measure those yet, so the first iteration can just be core ping stuff.

Reply:
I'm not sure how we were ever sending those to be honest.. look at the schema: https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/master/schemas/telemetry/core/core.7.schema.json

Reply:
How do I see what table that schema maps to?
The table `default.mobile_clients_v2` has columns
`default_new_tab_experience`  `default_mail_client`

Reply:
Can I leave these out for first landing?

Reply:
Its ok by me since I don't think we're relying on that data right now for anything. We should probably ask @st3fan though

Reply:
Good point, @st3fan I want to drop sending `default_new_tab_experience` and `default_mail_client`. We can add them back in 10. Are they being used *right now*? If not, we have more work to do to figure out what the story is behind those, for instance they aren't in the schema mentioned above.

Reply:
Confirmed `default_mail_client` is used, so that is precondition of a proper landing of this code.